openstack_exporter/collectors/cinderbackend.py

```diff
@@ -16,23 +16,51 @@
 import math
 from prometheus_client.core import GaugeMetricFamily, InfoMetricFamily
 
+from cachetools import cached, TTLCache
 from cinderclient import client as cinder
 from openstack_exporter import BaseCollector
 
 LOG = logging.getLogger('openstack_exporter.exporter')
 
+DEFAULT_VALID_SHARDING_BACKENDS = ['standard_hdd', 'vmware']
+DEFAULT_VALID_NO_SHARD_BACKENDS = []
+ALLOW_UNEXPECTED_BACKENDS_DEFAULT = False
+
 
 class CinderBackendCollector(BaseCollector.BaseCollector):
-    version = "1.0.2"
+    version = "1.0.3"
 
-    def __init__(self, openstack_config):
-        super().__init__(openstack_config)
+    def __init__(self, openstack_config, collector_config):
+        super().__init__(openstack_config, collector_config)
         # We have to create a cinder client here
         # because the openstacksdk doesn't currently
         # Support the quota functions.
         self.cinder_client = self._cinder_client()
         self.labels = ['backend', 'pool', 'shard']
         LOG.debug(f"Openstack Exporter CinderBackend Version {self.version}")
+        LOG.debug(f"Collector configuration {self.collector_config}")
+        # Make sure there is a default setting for valid_backends
+        cinder_config = self.collector_config['cinderbackend']
+        LOG.debug(f"Cinder_config {cinder_config}")
```

Comment:
Do we need to log `cinder_config` when we already logged `collector_config`, which should include `cinderbackend` i.e. `cinder_config` already?

Reply:
removed